pydrive2/fs/spec.py

```diff
@@ -326,13 +326,19 @@ def _get_remote_item_ids(self, parent_ids, title):
         )
         query += " and title='{}'".format(title.replace("'", "\\'"))
 
-        # GDrive list API is case insensitive, we need to compare
-        # all results and pick the ones with the right title
-        return [
-            item["id"]
-            for item in self._gdrive_list(query)
-            if item["title"] == title
-        ]
+        res = []
+        for item in self._gdrive_list(query):
+            # GDrive list API is case insensitive, we need to compare
+            # all results and pick the ones with the right title
+            if item["title"] == title:
+                res.append(item["id"])
+
+                if item["mimeType"] == FOLDER_MIME_TYPE:
+                    self._cache_path_id(
```

Comment:
C: This is the main change here. It was missing and DVC kept hitting `_get_remote_item_ids` with the same queries like `files/md5`, etc.